sigstore/_internal/trustroot.py

```diff
@@ -117,20 +279,27 @@ def _get_ca_keys(
             for cert in ca.cert_chain.certificates:
                 yield cert.raw_bytes
 
-    def get_ctfe_keys(self) -> list[bytes]:
-        """Return the CTFE public keys contents."""
-        ctfes: list[bytes] = list(self._get_tlog_keys(self.ctlogs))
-        if not ctfes:
-            raise MetadataError("CTFE keys not found in trusted root")
-        return ctfes
-
-    def get_rekor_keys(self) -> list[bytes]:
-        """Return the rekor public key content."""
-        keys: list[bytes] = list(self._get_tlog_keys(self.tlogs))
+    def rekor_keyring(self) -> RekorKeyring:
+        """Return keyring with keys for Rekor."""
 
+        keys: list[bytes]
+        if self.args and self.args.rekor_root_pubkey:
+            keys = self.args.rekor_root_pubkey.read()
+        else:
+            keys = list(self._get_tlog_keys(self.tlogs, self.purpose))
         if len(keys) != 1:
             raise MetadataError("Did not find one Rekor key in trusted root")
-        return keys
+        return RekorKeyring(Keyring(keys))
```

Comment:
This should work:

```suggestion
        return RekorKeyring(keys)
```

Reply:
I agree with you that it should work as RekorKeyring is a subtype of Keyring, but it doesn't :/